api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -156,6 +156,14 @@ message HttpService {
 
   // Settings used for controlling authorization response metadata.
   AuthorizationResponse authorization_response = 8;
+
+  // Runtime flag that controls whether to use lowercase string matcher or not for headers matching,
+  // for both patterns included in :ref:`authorization_request
+  // <envoy_api_field_config.filter.http.ext_authz.v2.HttpService.authorization_request>` and
+  // :ref:`authorization_response
+  // <envoy_api_field_config.filter.http.ext_authz.v2.HttpService.authorization_response>`. If not
+  // specified, defaults to use the lowercase string matcher.
+  api.v2.core.RuntimeFeatureFlag use_lowercase_string_matcher = 9;
```

Comment:
I don't think I would add an actual config field for this. Can you use the `RuntimeFeatures` support that @alyssawilk has added and just document the key to switch the behavior in the deprecated and release notes? IMO it's probably OK to just swap the behavior to disable this behavior with the runtime key there to turn it on but happy to discuss an inversion if you think it would be better.